libs/sdk-common/src/utils/rest_client.rs

```diff
@@ -1,84 +1,109 @@
 use log::*;
-use reqwest::StatusCode;
-use std::time::Duration;
+use reqwest::Client;
+use std::{collections::HashMap, time::Duration};
 
 use crate::error::{ServiceConnectivityError, ServiceConnectivityErrorKind};
 
 const REQUEST_TIMEOUT: Duration = Duration::from_secs(30);
 
-/// Creates an HTTP client with a built-in connection timeout
-pub fn get_reqwest_client() -> Result<reqwest::Client, ServiceConnectivityError> {
-    reqwest::Client::builder().build().map_err(Into::into)
-}
+#[sdk_macros::async_trait]
+pub trait RestClient: Send + Sync {
+    /// Makes a GET request and logs on DEBUG.
+    /// ### Arguments
+    /// - `url`: the URL on which GET will be called
+    /// - `enforce_status_check`: if true, the HTTP status code is checked in addition to trying to
+    ///    parse the payload. In this case, an HTTP error code will automatically cause this function to
+    ///    return `Err`, regardless of the payload. If false, the result type will be determined only
+    ///    by the result of parsing the payload into the desired target type.
+    async fn get_and_log_response(
+        &self,
+        url: &str,
+        enforce_status_check: bool,
+    ) -> Result<String, ServiceConnectivityError>;
 
-pub async fn post_and_log_response(
-    url: &str,
-    body: Option<String>,
-) -> Result<String, ServiceConnectivityError> {
-    debug!("Making POST request to: {url}");
+    /// Makes a POST request, and logs on DEBUG.
+    /// ### Arguments
+    /// - `url`: the URL on which POST will be called
+    /// - `headers`: the optional POST headers
+    /// - `body`: the optional POST body
+    async fn post_and_log_response(
```

Comment:
At the trait level I think it is better to make the function naming/definitions more simplified and less verbose.
IMO simply get & post are good options and we can keep the logging in the implementation without the need to express it in the function name.
I also think the result should be body and status code, letting the caller the option to enforce_status_check or not. I have no objection to add another function or two that gets the rest client as parameter and have more options just feel that the interface level (especially  because it is also exposed out of this trait) should be clean and minimal.

Reply:
I've renamed the functions and included the status code in the response [a395257](https://github.com/breez/breez-sdk-greenlight/pull/1175/commits/a395257c1544b989ec7ecb2b0c8487f3bd910229). 

To expose the client internals we probably need to add another trait, I don't see any use at the moment. Should we add it @roeierez ?